lib/features/modeling/behavior/CompensateBoundaryEventBehaviour.js

```diff
@@ -12,124 +12,105 @@ import { hasEventDefinition } from '../../../util/DiUtil';
 
 export default function CompensateBoundaryEventBehaviour(eventBus, modeling, bpmnRules) {
 
-  var replacing = false;
-
   CommandInterceptor.call(this, eventBus);
 
   function addIsForCompensationProperty(target) {
-    if (is(target, 'bpmn:Activity') && !isForCompensation(target)) {
-      modeling.updateProperties(target, { isForCompensation: true });
-
-    }
+    modeling.updateProperties(target, { isForCompensation: true });
   }
 
   function removeIsForCompensationProperty(target) {
-    if (is(target, 'bpmn:Activity') && isForCompensation(target)) {
-      modeling.updateProperties(target, { isForCompensation: false });
-
-    }
+    modeling.updateProperties(target, { isForCompensation: undefined });
   }
 
-  function removeIlegalConnections(element) {
+  function removeIlegalConnections(element, exception) {
```

Comment:
I would have expected something like `ignoredConnection(s)`. `Exception` is ambiguous. 💣